### ISSUE TITLE
[BUG] Symmetry effect warps on low poly meshes

### ISSUE THREAD

**Issue by jlampel** [CGCookie/retopoflow#916]
Blender version: 2.91.0
RetopoFlow: 3.00.2
Windows 10

Symmetry appears to have a display bug, and I don't know enough about OpenGL to guess at what it might be:
![TcS4vfU3gH](https://user-images.githubusercontent.com/12982300/101539206-00989300-3953-11eb-952c-7fdfd0ced06a.gif)

However, it only seems to happen on super simple objects. On high poly sculpts it seems to be just fine. Here's a good example of how the polygon density influences the behavior:
![I9TMP5OtFm](https://user-images.githubusercontent.com/12982300/101539873-0a6ec600-3954-11eb-859a-6595a63c9b75.gif)

Reproducible in any default file.

**Comment by vxlcoder** [comment]
this is due to the low resolution of the source mesh and how we render the symmetry visualization.  specifically, we use the vertex normals to offset the render so the symmetry viz appears "on top" of the source mesh.  the problem is that low res meshes have highly divergent normals, so the verts coincident to polygons that are facing away from user get warped.  (think about how these meshes would look if smooth shading were enabled... how the shading would look wrong)

in other words: these meshes are not target source meshes at the moment.

a workaround is to remesh or subdivide+decimate the source mesh.  or, you could add additional edge loops about those "backside" edges, similar to the edge loops that would be added to control subdivision if you didn't want to use edge creases.

**Comment by vxlcoder** [comment]
alternative: separate every face (do not share verts among the faces) in the source symmetry visualization, but this would ~4x memory for vertices, and would cost additional time to do.

**Comment by jlampel** [comment]
Adding this here just to keep things in the right thread:

> Would it make sense to use vertex-per-face normals instead so that it splits the faces that Blender considers sharp?

That should make it so that there's only more memory used if the object has flat shading or custom normals, which is exactly when it would be needed. It's not a big priority though as most sculpts in need of retopology are very dense and smooth shaded. It could be the case that someone tries to start RF with a dense but flat shaded source mesh, and in that case we could recommend via a note that they set it to smooth shading.

**Comment by jlampel** [comment]
Closing since this system is not used in v4
